lib/foreman/engine.rb

```diff
@@ -351,6 +354,14 @@ def flush_reader(reader)
 
 ## Engine ###########################################################
 
+  def check_processes
+    return unless formation.length > 0
+
+    process_name = formation.keys.first
+    process_name_exists = @names.values.include?(process_name)
```

Comment:
I can't really understand what this check is doing. If they specified a name at the command line we want to make sure it's an actual process.